ReactWindows/ReactNative/ReactInstanceManagerImpl.cs

```diff
@@ -0,0 +1,238 @@
+        {
+            foreach (var nativeModule in reactPackage.CreateNativeModules(reactContext))
+            {
+                nativeRegistryBuilder.Add(nativeModule);
+            }
+
+            foreach (var type in reactPackage.CreateJavaScriptModulesConfig())
+            {
+                jsModulesBuilder.Add(type);
+            }
+        }
+
+        /// <summary>
+        /// Attaches the <see cref="ReactRootView" /> to the list of tracked root views
+        /// </summary>
+        /// <param name="rootView">The root view for the ReactJS app</param>
+        public override void AttachMeasuredRootView(ReactRootView rootView)
+        {
+            _attachedRootViews.Add(rootView);
+            
+            if (var != null)
+            {
+                AttachMeasuredRootViewToInstance(rootView, var.CatalystInstance);
+            }
+        }
+
+        /// <summary>
+        /// Detach given <see cref="rootView" /> from current catalyst instance. 
+        /// </summary>
+        /// <param name="rootView">The root view for the ReactJS app</param>
+        public override void DetachRootView(ReactRootView rootView)
+        {
+            if (_attachedRootViews.Remove(rootView))
+            {
+                if (var != null)
+                {
+                    DetachViewFromInstance(rootView, var.CatalystInstance);
+                }
+            }
+        }
+
+        private void DetachViewFromInstance(ReactRootView rootView, ICatalystInstance catalystInstance)
+        {
+            try
+            {
+                catalystInstance.GetJavaScriptModule<AppRegistry>()?.unmountApplicationComponentAtRootTag(rootView.TagId);
+            }
+            catch (InvalidOperationException ex)
+            {
+                throw new InvalidOperationException("Unable to load AppRegistry JS module. Error message: " + ex.Message);
+            }
+        }
+
+        private void AttachMeasuredRootViewToInstance(ReactRootView rootView, ICatalystInstance catalystInstance)
+        {
+            UIManagerModule uiManagerModule = catalystInstance.GetNativeModule<UIManagerModule>();
+            int rootTag = uiManagerModule.AddMeasuredRootView(rootView);
+            var initialProps = new Dictionary<string, object>();
+                initialProps.Add("rootTag", rootTag);
+
+            try
+            {
+                catalystInstance.GetJavaScriptModule<AppRegistry>()?.runApplication(rootView.JSModuleName, initialProps);
+            }
+            catch (InvalidOperationException ex)
+            {
+                throw new InvalidOperationException("Unable to load AppRegistry JS module. Error message: " + ex.Message);
+            }
+        }
+
+        private void InvokeDefaultOnBackPressed()
+        {
+            DispatcherHelpers.AssertOnDispatcher();
+            // TODO: implement
+        }
+
+        class DefaultHardwareBackButtonHandlerImpl : IDefaultHardwareBackButtonHandler
```

Comment:
We should create an anonymous implementation that takes an Action in the constructor.